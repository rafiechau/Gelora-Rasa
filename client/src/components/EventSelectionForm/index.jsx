/* eslint-disable react/prop-types */
import React from 'react';
import PropTypes from 'prop-types';
import { FormControl, InputLabel, MenuItem, Select } from '@mui/material';
import { injectIntl, FormattedMessage } from 'react-intl';
import classes from './style.module.scss';

const EventSelectionForm = ({
  selectedEventId,
  allMyEvents,
  inputMeetingId,
  handleEventChange,
  handleMeetingIdChange,
  handleSubmit,
  intl: { formatMessage },
}) => (
  <form onSubmit={handleSubmit} className={classes.form} data-testid="event-selection-form">
    <FormControl fullWidth>
      <InputLabel id="demo-simple-select-label">Your Active Event</InputLabel>
      <Select
        labelId="demo-simple-select-label"
        id="demo-simple-select"
        value={selectedEventId}
        label={formatMessage({ id: 'app_your_event_placeholder' })}
        onChange={handleEventChange}
        data-testid="event-select"
      >
        {allMyEvents.map((event) => (
          <MenuItem key={event.id} value={event.id} data-testid={`event-option-${event.id}`}>
            {event.eventName}
          </MenuItem>
        ))}
      </Select>
    </FormControl>
    <input
      className={classes.inputMeetingId}
      placeholder={formatMessage({ id: 'app_streaming_send_meeting_id_to_customer' })}
      value={inputMeetingId}
      onChange={handleMeetingIdChange}
      data-testid="meeting-id-input"
    />
    <button className={classes.btnSendToUser} type="submit" data-testid="submit-button">
      <FormattedMessage id="app_streaming_send_to_customer" />
    </button>
  </form>
);

EventSelectionForm.propTypes = {
  allMyEvents: PropTypes.array.isRequired,
  inputMeetingId: PropTypes.string.isRequired,
  handleEventChange: PropTypes.func.isRequired,
  handleMeetingIdChange: PropTypes.func.isRequired,
  handleSubmit: PropTypes.func.isRequired,
  intl: PropTypes.object,
};

export default injectIntl(EventSelectionForm);
