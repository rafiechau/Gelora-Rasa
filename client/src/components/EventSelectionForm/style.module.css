.form {
  width: 90%;
  border: 1px solid #2978b5;
  padding: 20px;
  border-radius: 12px;
  box-shadow: 0 4px 8px rgba(0, 0, 0, 0.2);
  display: flex;
  flex-direction: column;
  gap: 1rem;
}
.form .inputMeetingId {
  height: 40px;
  border-radius: 8px;
  margin-bottom: 1rem;
}
.form input::-moz-placeholder {
  text-indent: 10px;
}
.form input::placeholder {
  text-indent: 10px;
}
.form .btnSendToUser {
  height: 40px;
  border-radius: 4px;
  margin-bottom: 1rem;
  background-color: #2978b5;
  color: #ffffff;
  padding: 10px 20px;
  font-size: 1rem;
  border: none;
  border-radius: 4px;
  cursor: pointer;
  transition: background-color 0.3s ease;
  height: 40px;
}
.form .btnSendToUser:hover {
  background-color: #205c8b;
}
@media (max-width: 480px) {
  .form .btnSendToUser {
    width: 100%;
    margin-bottom: 10px;
  }
}/*# sourceMappingURL=style.module.css.map */