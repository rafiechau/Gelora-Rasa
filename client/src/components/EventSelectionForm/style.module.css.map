{"version":3,"sources":["style.module.scss","style.module.css"],"names":[],"mappings":"AAKA;EACI,UAAA;EACA,yBAAA;EACA,aAAA;EACA,mBAAA;EACA,wCAAA;EACA,aAAA;EACA,sBAAA;EACA,SAAA;ACJJ;ADKI;EACE,YAAA;EACA,kBAAA;EACA,mBAAA;ACHN;ADKI;EACE,iBAAA;ACHN;ADEI;EACE,iBAAA;ACHN;ADKI;EACE,YAAA;EACF,kBAAA;EACA,mBAAA;EAEE,yBA1BS;EA2BT,cA1BO;EA2BP,kBAAA;EACA,eAAA;EACA,YAAA;EACA,kBAAA;EACA,eAAA;EACA,sCAAA;EAIA,YAAA;ACPN;ADIM;EACE,yBAjCM;AC+Bd;ADKM;EAjBF;IAkBI,WAAA;IACA,mBAAA;ECFN;AACF","file":"style.module.css"}