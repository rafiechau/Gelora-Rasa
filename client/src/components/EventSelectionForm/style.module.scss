$background-color: #151422;
$button-color: #2978b5; 
$text-color: #ffffff; 
$hover-color: darken($button-color, 10%);

.form{
    width: 90%;
    border: 1px solid $button-color;
    padding: 20px;
    border-radius: 12px;
    box-shadow: 0 4px 8px rgba(0, 0, 0, 0.2);
    display: flex;
    flex-direction: column;
    gap: 1rem;
    .inputMeetingId{
      height: 40px;
      border-radius: 8px; // Bentuk yang lebih modern
      margin-bottom: 1rem;
    }
    input::placeholder{
      text-indent: 10px;
    }
    .btnSendToUser{
      height: 40px;
    border-radius: 4px; 
    margin-bottom: 1rem;

      background-color: $button-color;
      color: $text-color;
      padding: 10px 20px;
      font-size: 1rem;
      border: none;
      border-radius: 4px;
      cursor: pointer;
      transition: background-color 0.3s ease;
      &:hover {
        background-color: $hover-color;
      }
      height: 40px; 
      @media (max-width: 480px) {
        width: 100%;
        margin-bottom: 10px;
      }
    }
  }