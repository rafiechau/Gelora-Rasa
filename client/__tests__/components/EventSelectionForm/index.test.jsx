// EventSelectionForm.test.jsx
import React from 'react';
import { render, fireEvent, screen } from '@testing-library/react';
import '@testing-library/jest-dom';
import { IntlProvider } from 'react-intl';
import EventSelectionForm from '@components/EventSelectionForm';

const messages = {
  app_your_event_placeholder: 'Your Active Event',
  app_streaming_send_meeting_id_to_customer: 'Enter Meeting ID',
  app_streaming_send_to_customer: 'Send to Customer',
};

describe('EventSelectionForm', () => {
  const mockEvents = [
    { id: '1', eventName: 'Event 1' },
    { id: '2', eventName: 'Event 2' },
  ];
  const mockHandleEventChange = jest.fn();
  const mockHandleMeetingIdChange = jest.fn();
  const mockHandleSubmit = jest.fn();

  beforeEach(() => {
    render(
      <IntlProvider locale="en" messages={messages}>
        <EventSelectionForm
          selectedEventId="1"
          allMyEvents={mockEvents}
          inputMeetingId=""
          handleEventChange={mockHandleEventChange}
          handleMeetingIdChange={mockHandleMeetingIdChange}
          handleSubmit={mockHandleSubmit}
        />
      </IntlProvider>
    );
  });

  test('Should render correctly', () => {
    expect(screen.getByTestId('event-selection-form')).toBeInTheDocument();
  });
  test('Should call onSubmit when form is submitted', () => {
    fireEvent.click(screen.getByTestId('submit-button'));
    expect(mockHandleSubmit).toHaveBeenCalled();
  });

  test('Should match snapshot', () => {
    const form = screen.getByTestId('event-selection-form');
    expect(form).toMatchSnapshot();
  });
});
