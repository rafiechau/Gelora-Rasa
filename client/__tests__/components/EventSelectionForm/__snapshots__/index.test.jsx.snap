// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EventSelectionForm Should match snapshot 1`] = `
<form
  class="form"
  data-testid="event-selection-form"
>
  <div
    class="MuiFormControl-root MuiFormControl-fullWidth css-q8hpuo-MuiFormControl-root"
  >
    <label
      class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiFormLabel-filled MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined css-1jy569b-MuiFormLabel-root-MuiInputLabel-root"
      data-shrink="true"
      id="demo-simple-select-label"
    >
      app_event_active
    </label>
    <div
      class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl  css-1yk1gt9-MuiInputBase-root-MuiOutlinedInput-root-MuiSelect-root"
      data-testid="event-select"
    >
      <div
        aria-expanded="false"
        aria-haspopup="listbox"
        aria-labelledby="demo-simple-select-label demo-simple-select"
        class="MuiSelect-select MuiSelect-outlined MuiInputBase-input MuiOutlinedInput-input css-11u53oe-MuiSelect-select-MuiInputBase-input-MuiOutlinedInput-input"
        id="demo-simple-select"
        role="button"
        tabindex="0"
      >
        Event 1
      </div>
      <input
        aria-hidden="true"
        aria-invalid="false"
        class="MuiSelect-nativeInput css-yf8vq0-MuiSelect-nativeInput"
        tabindex="-1"
        value="1"
      />
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiSelect-icon MuiSelect-iconOutlined css-hfutr2-MuiSvgIcon-root-MuiSelect-icon"
        data-testid="ArrowDropDownIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M7 10l5 5 5-5z"
        />
      </svg>
      <fieldset
        aria-hidden="true"
        class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
      >
        <legend
          class="css-14lo706"
        >
          <span>
            Your Active Event
          </span>
        </legend>
      </fieldset>
    </div>
  </div>
  <input
    class="inputMeetingId"
    data-testid="meeting-id-input"
    placeholder="Enter Meeting ID"
    value=""
  />
  <button
    class="btnSendToUser"
    data-testid="submit-button"
    type="submit"
  >
    Send to Customer
  </button>
</form>
`;
